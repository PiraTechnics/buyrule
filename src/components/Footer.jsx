import Brand from "./Brand";

const Footer = () => {
  return (
    <footer className="bg-sky-600 w-screen py-2 mx-auto px-2 sm:px-6 lg:px-8">
      <div className="relative flex justify-center gap-16">
        <Brand />
        <div className="text-sm font-hylia self-end">
          <span>Â© Devin Younge</span> <span>{new Date().getFullYear()}</span>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
