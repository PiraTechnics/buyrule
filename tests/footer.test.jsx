import { render, screen } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import Footer from "/src/components/Footer";

beforeEach(() => {
  render(<Footer />);
});

describe("Footer Component", () => {
  it("contains copyright info", () => {
    expect(screen.getByText("Â© Devin Younge")).toBeInTheDocument();
  });

  it("hasn't changed unexpectedly", () => {
    // since screen does not have the container property, we'll destructure render to obtain a container for this test
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
