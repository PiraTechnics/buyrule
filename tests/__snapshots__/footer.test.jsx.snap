// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > hasn't changed unexpectedly 1`] = `
<div>
  <footer
    class="bg-sky-600 w-screen py-2 mx-auto px-2 sm:px-6 lg:px-8"
  >
    <div
      class="relative flex justify-center gap-16"
    >
      <div
        class="text-4xl text-white tracking-logo font-hylia"
      >
        <div
          class="-mb-2 flex flex-row"
        >
          BUY
          <img
            alt="rupee icon"
            class="w-8 pt-2 ms-1"
            src="/src/assets/rupee-logo-tilted.svg"
          />
        </div>
        <div
          class=""
        >
          RULE
        </div>
      </div>
      <div
        class="text-sm font-hylia self-end"
      >
        <span>
          Â© Devin Younge
        </span>
         
        <span>
          2023
        </span>
      </div>
    </div>
  </footer>
</div>
`;
